src/Illuminate/Support/helpers.php

```diff
@@ -337,15 +337,15 @@ function object_get($object, $key, $default = null)
      * Provide access to optional objects.
      *
      * @param  mixed  $value
-     * @param  callable|null  $callback
+     * @param  mixed  $second
      * @return mixed
      */
-    function optional($value = null, callable $callback = null)
+    function optional($value = null, $second = null)
     {
-        if (is_null($callback)) {
-            return new Optional($value);
+        if (! is_callable($second)) {
```

Comment:
```php
if (is_callable($second)) {
    return $second($value);
}

return new Optional($value, $second);
```

Reply:
The callback is only callled when the object is not `null` I can make the condition `is_callable($second) && !is_null($value)` anyways. Thanks for the suggestion.